<launch>

  <arg name="bag_name" default="tmp" />

  <include file="$(find racer)/launch/ydlidar.launch" />
  <include file="$(find racer)/launch/arduinos.launch" />

  <node pkg="tf" type="static_transform_publisher" name="base_link_2_laser_frame" args="0.125 0 0.135 0 0 0 base_link laser_frame 20" />

  <node pkg="hector_mapping" type="hector_mapping" name="hector_mapping" output="screen">
    <param name="scan_topic" value="scan" />
    <param name="map_frame" value="map" />
    <param name="base_frame" value="base_link" />
    <param name="odom_frame" value="base_link" />
    <param name="output_timing" value="false"/>
    <param name="advertise_map_service" value="true"/>
    <param name="use_tf_scan_transformation" value="true"/>
    <param name="use_tf_pose_start_estimate" value="false"/>
    <param name="pub_map_odom_transform" value="true"/>
    <param name="map_with_known_poses" value="false"/>
    <param name="map_pub_period" value="2"/>
    <param name="update_factor_free" value="0.45"/>
    <param name="map_update_distance_thresh" value="0.02"/>
    <param name="map_update_angle_thresh" value="0.1"/>
    <param name="map_resolution" value="0.04"/>
    <param name="map_size" value="512"/>
    <param name="map_start_x" value="0.5"/>
    <param name="map_start_y" value="0.5"/>
  </node>

  <node pkg="rosbag" type="record" name="record" args="-O $(find racer)/launch/recorded-bags/$(arg bag_name).bag /scan /racer/commands /racer/wheel_odometry" />

  <node name="rviz" pkg="rviz" type="rviz" args="-d $(find racer)/rviz/mapping.rviz" />

 </launch>
